codalab/worker/dependency_manager.py

```diff
@@ -53,6 +98,9 @@ class DependencyManager(StateTransitioner, BaseDependencyManager):
     # the data format of how we store this)
     MAX_SERIALIZED_LEN = 60000
 
+    # If it has been this long since a worker has downloaded anything, another worker will take over downloading.
```

Comment:
Do additional testing:
- run bundles on workers A, B, C that all share cache
- force kill worker A while it's downloading a dependency
- see how long it takes the dependency to be re-downloaded by B or C

Reply:
Did you do this yet @teetone ?